src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyArray = [];
+  let changedObject = { ...state };
```

Comment:
The name `changedObject` for the copy of the state is not descriptive enough. Consider renaming it to `stateCopy` or `currentState` to better reflect its purpose.